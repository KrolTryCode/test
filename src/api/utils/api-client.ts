import axios, { AxiosError, InternalAxiosRequestConfig } from 'axios';
import i18n from 'i18next';
import queryString from 'query-string';

import { ApiConfig, Api } from '~/api/utils/api-requests';
import { authPath, loginPath } from '~/utils/configuration/routes-paths';
import {
  projectLocalStorageService,
  projectSessionStorageService,
} from '~/utils/localstorage/project-storage/project-storage-instance';
import { ProjectStorageKey } from '~/utils/localstorage/project-storage/project-storage.types';
import { showErrorMessage } from '~/utils/show-error-message';

export interface ErrorResponse {
  code: number;
  message: string;
}

const getStorageService = () => {
  const rememberMe = projectLocalStorageService.get(ProjectStorageKey.RememberMe);
  return rememberMe ? projectLocalStorageService : projectSessionStorageService;
};

const BASE_API_URL = '/api';

const API_CONFIG: ApiConfig = {
  baseURL: BASE_API_URL,
};

const API_CONFIG_SECURE: ApiConfig = {
  baseURL: BASE_API_URL,
  securityWorker: async () => {
    const token = getStorageService().get(ProjectStorageKey.AccessToken);

    if (token) {
      return { headers: { Authorization: `Bearer ${token ?? ''}` } };
    }
  },
  paramsSerializer: params => {
    const keys = Object.keys(params);

    let res: {
      [key: string]: unknown;
    } = {};

    for (const key of keys) {
      if (typeof params[key] === 'object' && !Array.isArray(params[key])) {
        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
        res = { ...res, ...params[key] };
      } else {
        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
        res[key] = params[key];
      }
    }

    return queryString.stringify(res, { arrayFormat: 'none' });
  },
};

export const ApiClient = new Api(API_CONFIG);

export const ApiClientSecured = new Api(API_CONFIG_SECURE);

ApiClientSecured.instance.interceptors.request.use(request => {
  request.headers.set('Accept-Language', i18n.language);
  return request;
});

ApiClient.instance.interceptors.request.use(request => {
  request.headers.set('Accept-Language', i18n.language);
  return request;
});

// https://gist.github.com/Godofbrowser/bf118322301af3fc334437c683887c5f#file-axios-refresh_token-1-js
let isRefreshing = false;
let failedQueue: {
  resolve: (token: string | null) => void;
  reject: (error: AxiosError) => void;
}[] = [];

const processQueue = (error: AxiosError | null, token: string | null = null) => {
  failedQueue.forEach(prom => {
    error ? prom.reject(error) : prom.resolve(token);
  });
  failedQueue = [];
};

// Можно задать кастомные обработчики для каждой ошибки
ApiClientSecured.instance.interceptors.response.use(
  response => response,
  async (error: AxiosError) => {
    const originalRequest = error.config as InternalAxiosRequestConfig<any> & { _retry: boolean };

    if (error.response?.status === 502) {
      showErrorMessage(error, 'ERROR.TECHNICAL_ERROR');
    }
    if (error.response?.status === 401 && !originalRequest._retry) {
      if (error.config?.url === '/api/v1/jwt/refresh') {
        window.location = `/${authPath}/${loginPath}` as unknown as Location;
        return;
      }

      if (window.location.pathname.startsWith(`/${authPath}`)) {
        return Promise.reject(error);
      }

      if (isRefreshing) {
        return new Promise(function (resolve, reject) {
          failedQueue.push({ resolve, reject });
        })
          .then(token => {
            originalRequest.headers['Authorization'] = 'Bearer ' + String(token);
            return axios(originalRequest);
          })
          .catch(err => {
            return Promise.reject(err);
          });
      }
      originalRequest._retry = true;
      isRefreshing = true;

      const storage = getStorageService();
      storage.remove(ProjectStorageKey.AccessToken);
      const refreshToken = storage.get(ProjectStorageKey.RefreshToken);

      return new Promise(function (resolve, reject) {
        ApiClientSecured.tokenV1Controller
          .regenerateAccessToken({ headers: { Authorization: `Bearer ${refreshToken ?? ''}` } })
          .then(({ token }) => {
            storage.set(ProjectStorageKey.AccessToken, token!);
            axios.defaults.headers.common['Authorization'] = 'Bearer ' + token;
            processQueue(null, token);
            if (originalRequest) {
              originalRequest.headers['Authorization'] = 'Bearer ' + token;
              resolve(axios(originalRequest));
            }
          })
          .catch((err: AxiosError) => {
            processQueue(err, null);
            reject(err);
          })
          .finally(() => {
            isRefreshing = false;
          });
      });
    }
    return Promise.reject(error);
  },
);
