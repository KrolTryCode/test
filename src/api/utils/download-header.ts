import { RequestParams } from './api-requests';

export const getDownloadHeaderCSV = (): RequestParams => ({
  headers: {
    accept: 'text/csv',
  },
});
