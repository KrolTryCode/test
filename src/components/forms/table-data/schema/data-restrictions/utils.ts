import {
  ColumnValue,
  Const,
  FloatConst,
  IntConst,
  StringConst,
  StringLengthValue,
} from '~/api/utils/api-requests';

export const isColumnValueType = (value: any) => {
  return '@type' in value && value['@type'] === 'ColumnValue';
};

export const getRightOrLeftValue = (
  currentObject:
    | string
    | object
    | ColumnValue
    | Const
    | StringLengthValue
    | FloatConst
    | IntConst
    | StringConst,
) => {
  let value: object | string;
  if (typeof currentObject === 'object' && 'value' in currentObject) {
    value = currentObject.value;
  } else {
    value = currentObject;
  }
  return value;
};
