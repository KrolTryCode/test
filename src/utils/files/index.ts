export { createObjectURLFromFile } from './create-object-url-from-file';
export { downloadBlobFile } from './download-blob-file';
export { downloadCsvFile } from './download-csv-file';
