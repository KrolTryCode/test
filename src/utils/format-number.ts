import roundToFixed from 'round-tofixed';

export function formatNumberNoTrailingZeros(number: unknown, precision = 2) {
  if (typeof number !== 'number') {
    return '';
  }

  return roundToFixed(number, precision);
}
