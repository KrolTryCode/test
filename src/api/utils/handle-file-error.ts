import { AxiosError } from 'axios';

import { showErrorMessage } from '~/utils/show-error-message';

export function handleFileUploadingError(e: unknown) {
  if (e instanceof AxiosError && e.response?.status === 413) {
    showErrorMessage(e, 'FILES.ERROR.TOO_BIG');
  } else {
    showErrorMessage(e, 'FILES.UPLOAD_FAILED');
  }
}
