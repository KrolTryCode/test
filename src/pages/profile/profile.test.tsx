import { screen } from '@testing-library/react';
import { it, describe, expect } from 'vitest';

import { mockedParams } from 'tests/__mocks__/react-router-dom.mock';
import { renderComponent } from 'tests/utils/render-component';
import Profile from '~/pages/profile/profile.page';

describe('Profile', () => {
  beforeAll(() => {
    mockedParams.mockReturnValue({ userId: 'test-admin-id-2c963f66afa6' });
    renderComponent(<Profile />, { withRouter: true });
  });

  it('renders the profile', () => {
    const container = screen.getByTestId('appContainer');
    expect(container).toMatchSnapshot('profile');
  });
});
