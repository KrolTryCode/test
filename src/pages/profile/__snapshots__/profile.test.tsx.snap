// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Profile > renders the profile > profile 1`] = `
<div
  data-testid="appContainer"
>
  <h2
    class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-10p8o4e-MuiTypography-root"
  >
    Профиль
  </h2>
  <div
    class="MuiStack-root css-1on8rii-MuiStack-root"
  >
    <div
      class="MuiStack-root css-wqgdpk-MuiStack-root"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-7e98t4-MuiAvatar-root"
        color="primary"
      />
    </div>
    <form
      class="MuiStack-root css-ox3dtq-MuiStack-root"
      role="form"
    >
      <fieldset
        class="MuiStack-root css-106nt1m-MuiStack-root"
      >
        <legend
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-xgouly-MuiTypography-root"
        >
          <div
            class="MuiStack-root css-13tplj0-MuiStack-root"
          >
            Персональные данные
          </div>
        </legend>
        <fieldset
          class="MuiStack-root css-1javh08-MuiStack-root"
        >
          <div
            class="MuiFormControl-root css-k8zdik-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1xtmzjf-MuiFormLabel-root"
            >
              Фамилия
            </label>
            <div
              class="MuiBox-root css-eal7pa"
            >
              <div
                class="MuiBox-root css-0"
                data-mui-internal-clone-element="true"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-igw2tk-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-19kxmhd-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      aria-readonly="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1thzzby-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r1:"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-ihdtdm"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-k8zdik-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1xtmzjf-MuiFormLabel-root"
            >
              Имя
            </label>
            <div
              class="MuiBox-root css-eal7pa"
            >
              <div
                class="MuiBox-root css-0"
                data-mui-internal-clone-element="true"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-igw2tk-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-19kxmhd-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      aria-readonly="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1thzzby-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r3:"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-ihdtdm"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-k8zdik-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1xtmzjf-MuiFormLabel-root"
            >
              Отчество
            </label>
            <div
              class="MuiBox-root css-eal7pa"
            >
              <div
                class="MuiBox-root css-0"
                data-mui-internal-clone-element="true"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-igw2tk-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-19kxmhd-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      aria-readonly="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1thzzby-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r5:"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-ihdtdm"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </fieldset>
        <fieldset
          class="MuiStack-root css-1javh08-MuiStack-root"
        >
          <div
            class="MuiFormControl-root css-k8zdik-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1xtmzjf-MuiFormLabel-root"
            >
              Предприятие
            </label>
            <div
              class="MuiBox-root css-eal7pa"
            >
              <div
                class="MuiBox-root css-0"
                data-mui-internal-clone-element="true"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-igw2tk-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-19kxmhd-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      aria-readonly="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1thzzby-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r7:"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-ihdtdm"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-k8zdik-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1xtmzjf-MuiFormLabel-root"
            >
              Подразделение
            </label>
            <div
              class="MuiBox-root css-eal7pa"
            >
              <div
                class="MuiBox-root css-0"
                data-mui-internal-clone-element="true"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-igw2tk-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-19kxmhd-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      aria-readonly="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1thzzby-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r9:"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-ihdtdm"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-k8zdik-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1xtmzjf-MuiFormLabel-root"
            >
              Должность
            </label>
            <div
              class="MuiBox-root css-eal7pa"
            >
              <div
                class="MuiBox-root css-0"
                data-mui-internal-clone-element="true"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-igw2tk-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-19kxmhd-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      aria-readonly="false"
                      autocomplete="off"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1thzzby-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":rb:"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-ihdtdm"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </fieldset>
      </fieldset>
      <fieldset
        class="MuiStack-root css-106nt1m-MuiStack-root"
      >
        <legend
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-xgouly-MuiTypography-root"
        >
          <div
            class="MuiStack-root css-13tplj0-MuiStack-root"
          >
            Контакты
          </div>
        </legend>
        <fieldset
          class="MuiStack-root css-1javh08-MuiStack-root"
        >
          <div
            class="MuiFormControl-root css-k8zdik-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1xtmzjf-MuiFormLabel-root"
            >
              E-mail
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk css-ylp2hm-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiBox-root css-eal7pa"
            >
              <div
                class="MuiBox-root css-0"
                data-mui-internal-clone-element="true"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-igw2tk-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-19kxmhd-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      aria-readonly="false"
                      autocomplete="email"
                      class="MuiInputBase-input MuiOutlinedInput-input css-1thzzby-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":rd:"
                      type="email"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-ihdtdm"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </fieldset>
        <fieldset
          class="MuiStack-root css-1javh08-MuiStack-root"
        >
          <div
            class="MuiFormControl-root css-k8zdik-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1xtmzjf-MuiFormLabel-root"
            >
              Номер телефона
            </label>
            <div
              class="MuiBox-root css-eal7pa"
            >
              <div
                class="MuiBox-root css-0"
                data-mui-internal-clone-element="true"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root MuiTelInput-TextField css-igw2tk-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-1yk17tg-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-74b6xz-MuiInputAdornment-root"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                      <button
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-label="Select country"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiTelInput-IconButton css-10o7c2b-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiTelInput-Flag css-70qvj9"
                          data-testid="RU"
                        >
                          <img
                            alt="Russia"
                            class="MuiTelInput-FlagImg"
                            loading="lazy"
                            src="https://flagcdn.com/w40/ru.webp"
                            width="26"
                          />
                        </span>
                      </button>
                    </div>
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-z0ts0m-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":rf:"
                      type="tel"
                      value="+7"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-ihdtdm"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </fieldset>
      </fieldset>
      <fieldset
        class="MuiStack-root css-106nt1m-MuiStack-root"
      >
        <legend
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-xgouly-MuiTypography-root"
        >
          <div
            class="MuiStack-root css-13tplj0-MuiStack-root"
          >
            Системные данные
          </div>
        </legend>
        <fieldset
          class="MuiStack-root css-1javh08-MuiStack-root"
        >
          <div
            class="MuiFormControl-root css-k8zdik-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1xtmzjf-MuiFormLabel-root"
            >
              Дата регистрации
            </label>
            <div
              class="MuiBox-root css-eal7pa"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1mh8opi-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd Mui-readOnly MuiInputBase-readOnly css-iwzgqv-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd Mui-readOnly MuiInputBase-readOnly css-1u6kppm-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":rg:"
                    inputmode="text"
                    placeholder="ДД.ММ.ГГГГ чч:мм"
                    readonly=""
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1cj4z8e-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="Выберите дату"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-edgeEnd MuiIconButton-sizeSmall css-1x2tsee-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CalendarIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                        />
                      </svg>
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-k8zdik-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1xtmzjf-MuiFormLabel-root"
            >
              Дата последнего входа
            </label>
            <div
              class="MuiBox-root css-eal7pa"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1mh8opi-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd Mui-readOnly MuiInputBase-readOnly css-iwzgqv-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd Mui-readOnly MuiInputBase-readOnly css-1u6kppm-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":ri:"
                    inputmode="text"
                    placeholder="ДД.ММ.ГГГГ чч:мм"
                    readonly=""
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1cj4z8e-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="Выберите дату"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-edgeEnd MuiIconButton-sizeSmall css-1x2tsee-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CalendarIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                        />
                      </svg>
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-k8zdik-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1xtmzjf-MuiFormLabel-root"
            >
              IP-адрес последнего входа
            </label>
            <div
              class="MuiBox-root css-eal7pa"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-igw2tk-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-19kxmhd-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    aria-readonly="true"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1thzzby-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":rk:"
                    readonly=""
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-k8zdik-MuiFormControl-root"
            hidden=""
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1xtmzjf-MuiFormLabel-root"
            >
              Состояние
            </label>
            <div
              class="MuiBox-root css-eal7pa"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-igw2tk-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-19kxmhd-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    aria-readonly="true"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1thzzby-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":rl:"
                    readonly=""
                    type="text"
                    value="Нет данных"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </fieldset>
      </fieldset>
      <div
        class="MuiStack-root css-chek0q-MuiStack-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary MuiButton-disableElevation MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary MuiButton-disableElevation css-wg7wa7-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          id=":rm:"
          tabindex="0"
          type="button"
        >
          Изменить пароль
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-83fcih-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          id=":rn:"
          tabindex="0"
          type="submit"
        >
          Сохранить
        </button>
      </div>
    </form>
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-megsf6-MuiBackdrop-root"
      style="opacity: 0; visibility: hidden;"
    >
      <div
        class="MuiStack-root css-19rwzzo-MuiStack-root"
      >
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-82r84f-MuiCircularProgress-root"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate MuiCircularProgress-circleDisableShrink css-79nvmn-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1oh53cw-MuiTypography-root"
        >
          Загрузка...
        </p>
      </div>
    </div>
  </div>
</div>
`;
